lib/model.js

```diff
@@ -468,7 +468,11 @@ function addAuthToUrl({ url, isPasswordProtected }) {
   // so a single string replace should always work.
   url = url.replace('AUTH_TOKEN', authField, 1);
 
-  if (includes(['LDAP', 'KERBEROS', 'X509'], this.authStrategy)) {
+  if (
+    includes(['LDAP', 'KERBEROS', 'X509'], this.authStrategy)
+    && !url.includes('authSource=')
```

Comment:
this would sometimes apply `authSource=$external` twice to connection strings. Maybe a bug? Some of the tests had test cases with the duplicate `authSource=$external`. Not sure if intentional.